pkg/volume/plugins_test.go

```diff
@@ -165,3 +169,62 @@ func Test_ValidatePodTemplate(t *testing.T) {
 		t.Errorf("isPodTemplateValid(%v) returned (%v), want (%v)", pod.String(), got, "Error: pod specification does not contain any volume(s).")
 	}
 }
+
+// TestVolumePluginMultiThreaded tests FindPluginByName/FindPluginBySpec in a multi-threaded environment.
+// If these are called by different threads at the same time, they should still be able to reconcile the plugins
+// and return the same results (no missing plugin)
+func TestVolumePluginMultiThreaded(t *testing.T) {
```

Comment:
oops, not this test TestProberMultiThreaded

Reply:
revert in https://github.com/kubernetes/kubernetes/pull/128450

Reply:
ah I believe I need `wg.Add(1)` before the go routine starts, in case the go routine is very fast and `defer wg.Done()` is called before `wg.Add(1)` (I have only run this locally so it's probably not as fast on my laptop)
I can PR this change

Reply:
yes... add *always* has to be called before starting the goroutine that calls Done() ... I see that change needed in  several places

Reply:
Here's the fix: https://github.com/kubernetes/kubernetes/pull/128452

Reply:
Here's another fix PR (created before coming here): https://github.com/kubernetes/kubernetes/pull/128467

Reply:
I found a much more simple way to accomplish this PR: https://github.com/kubernetes/kubernetes/pull/128495